llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1528,6 +1528,24 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
         } else if (CallInst *const callInst = dyn_cast<CallInst>(value)) {
           if (callInst->getCalledFunction()->getName().startswith(gSPIRVMD::NonUniform))
             continue;
+        } else if (CmpInst *const cmpInst = dyn_cast<CmpInst>(value)) {
+          Value *lhs = nullptr;
+          Value *rhs = nullptr;
+          if (valueMap.count(cmpInst->getOperand(0))) {
+            lhs = valueMap[cmpInst->getOperand(0)];
+            rhs = cmpInst->getOperand(1);
+          } else if (valueMap.count(cmpInst->getOperand(1))) {
+            lhs = cmpInst->getOperand(0);
+            rhs = valueMap[cmpInst->getOperand(1)];
+          } else {
+            llvm_unreachable("Should never be called!");
+          }
```

Comment:
Yes, you are right and thanks, because I found one case which leads to assertion in current PR. 
I can think about two cases where both CmpInst operands are in valueMap

1) we are comparing two different row major matrices - this is not an issue because this will be handled in two different loop iterations. Each CmpInst operand will be from different `spirv.launder.row_major.X`
2) we are comparing the same row major matrices (comparing the same addresses) - this will lead to an assert.

I updated PR with code and tests.